lib/ansible/plugins/action/win_reboot.py

```diff
@@ -25,18 +25,34 @@ class TimedOutException(Exception):
 class ActionModule(ActionBase):
     TRANSFERS_FILES = False
 
-    DEFAULT_SHUTDOWN_TIMEOUT = 600
     DEFAULT_REBOOT_TIMEOUT = 600
     DEFAULT_CONNECT_TIMEOUT = 5
     DEFAULT_PRE_REBOOT_DELAY = 2
     DEFAULT_POST_REBOOT_DELAY = 0
     DEFAULT_TEST_COMMAND = 'whoami'
     DEFAULT_REBOOT_MESSAGE = 'Reboot initiated by Ansible.'
 
+    def get_system_uptime(self):
+        uptime_command = '''
```

Comment:
`(Get-WmiObject Win32_OperatingSystem).LastBootUpTime` is *probably* cheaper since you won't have to create a dynamic type on each invocation... All we really care about is if the value changed, I think, so we can probably treat it as opaque (ie, no need to parse it).

Reply:
Darn it is quicker, I must have tested the times when I already loaded the code but not in a blank session, will change it.